webviews/components/history_view.svelte

```diff
@@ -1,16 +1,175 @@
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT:
+        duration = +(tsToDate(history.eventTime).getTime() / 1000 - startTime).toFixed(3)
+        // console.log((tsToDate(history.eventTime).getTime()/1000),"-",startTime,"=",duration)
+        return `⇨ ${eventTypeName}`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_CONTINUED_AS_NEW:
+        duration = +(tsToDate(history.eventTime).getTime() / 1000 - startTime).toFixed(3)
+        // console.log((tsToDate(history.eventTime).getTime()/1000),"-",startTime,"=",duration)
+        return `⇨ ${eventTypeName}`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_ACTIVITY_TASK_SCHEDULED:
+        return ` ⇦ ${eventTypeName} (${history.activityTaskScheduledEventAttributes?.activityType?.name})`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_ACTIVITY_TASK_STARTED:
+        if (history.activityTaskStartedEventAttributes?.lastFailure) {
+          return ` ⇦ ${eventTypeName} (${history.activityTaskStartedEventAttributes?.lastFailure.message})`
+        } else {
+          return ` ⇦ ${eventTypeName}`
+        }
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_SIGNALED:
+        return ` ⇨ ${eventTypeName} (${history.workflowExecutionSignaledEventAttributes?.signalName})`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_TIMER_STARTED:
+        return `⇨ ⏱ ${eventTypeName} (${history.timerStartedEventAttributes?.startToFireTimeout?.seconds}s)`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_TIMER_FIRED:
+        return `⇦ ⏱ ${eventTypeName}`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_ACTIVITY_TASK_FAILED:
+        return ` ⇦ ${eventTypeName} (${history.activityTaskFailedEventAttributes?.failure})`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED:
+        duration = +(tsToDate(history.eventTime).getTime() / 1000 - startTime).toFixed(3)
+        // console.log((tsToDate(history.eventTime).getTime()/1000),"-",startTime,"=",duration)
+        return `⇨ ${eventTypeName}`
+    }
+    return `⇦ ${eventTypeName}`
+  }
 </script>
 
-<section>
-  <h1>[ ]Workflow Task</h1>
-
-  <p>
-    type: {history.events?.[0].eventType}
-    {history.events?.[0].workflowExecutionStartedEventAttributes?.workflowType?.name}
-  </p>
-  <p>Command out 1</p>
-  <p>Event in 2</p>
-  <p>Command out 2</p>
-</section>
+<head>
+  <title>Project Workflow panel</title>
+</head>
+
+<body>
+  <div>
+    {#if currentEvent === 2}
+      <h1>Workflow execution completed in {duration}s</h1>
+    {:else if currentEvent === 3}
+      <h1>Workflow execution failed in {duration}s</h1>
+    {:else if currentEvent === 4}
+      <h1>Workflow execution timed out after {duration}s</h1>
+    {:else if currentEvent === 27}
+      <h1>Workflow execution terminated after {duration}s</h1>
+    {:else if currentEvent === 28}
+      <h1>Workflow execution continued-as-new in {duration}s</h1>
+    {/if}
```

Comment:
Use constants instead (look at the enum definitions), and refactor this to a function please.